tests/trainer/test_trainer.py

```diff
@@ -1616,12 +1616,42 @@ def test_pytorch_profiler_trainer_ddp(tmpdir, use_output_filename):
 
     profiler = PyTorchProfiler(output_filename=output_filename)
 
+    model = BoringModel()
+    trainer = Trainer(fast_dev_run=True, profiler=profiler, accelerator="ddp", gpus=2, logger=TensorBoardLogger(tmpdir))
+    trainer.fit(model)
+
+    enabled = use_output_filename or not use_output_filename and profiler.local_rank == 0
+
+    if enabled:
+        assert len(profiler.summary()) > 0
+        assert set(profiler.profiled_actions.keys()) == {'training_step_and_backward', 'validation_step'}
+    else:
+        assert profiler.summary() is None
+        assert set(profiler.profiled_actions.keys()) == set()
+
+    if use_output_filename:
+        profiler.describe()
+        data = Path(profiler.output_fname).read_text()
+        assert len(data) > 0
+
+
+@pytest.mark.parametrize("use_output_filename", [True])
+def test_pytorch_profiler_trainer(tmpdir, use_output_filename):
+    """Ensure that the profiler can be given to the training and default step are properly recorded. """
+
+    if use_output_filename:
+        output_filename = os.path.join(tmpdir, "profiler.txt")
+    else:
+        output_filename = None
+
+    profiler = PyTorchProfiler(output_filename=output_filename)
+
     model = BoringModel()
     trainer = Trainer(
-        fast_dev_run=True,
+        max_epochs=1,
```

Comment:
@tchaton For regression tests please add tests for GPU and accelerator='ddp' as well

Reply:
Dear @chauhang, 

We don't against PyTorch 1.8 yet. 
We were planning to do so as soon as it is out.

I am going to check if we can test against 1.8.0 RC in the meanwhile: https://github.com/PyTorchLightning/pytorch-lightning/pull/6291

Best,
T.C